### PR TITLE
Sorting plants

### PR DESCRIPTION
**Pull request by chen4709**
- sort plants by bed and cultivar on homepage
- sort plants by common name and cultivar on bedpage

### CODE REVIEW

**Comment by chen4709** [comment]
Do not merge it yet.

**Comment by frogrammer** [comment]
We can merge this now.

File: server/src/main/java/umm3601/digitalDisplayGarden/PlantController.java [hunk 1]
```diff
@@ -505,4 +518,76 @@ public boolean addVisit(String plantID, String uploadID) {
         return null != plantCollection.findOneAndUpdate(filterDoc, push("metadata.visits", visit));
     }
 
+    /**
+     * Returns a Bson object generated by the given ArrayList of sorting fieldnames.
+     * The Bson object sorts by the criteria in ascending order.
+     * @param criteria - an ArrayList of Strings which are sorting filters
+     * @return a Bson object with sorting criteria
+     */
+    private Bson sortBy(ArrayList<String> criteria) {
```
Comment (lverns) [inline server/src/main/java/umm3601/digitalDisplayGarden/PlantController.java]: Is there any reason to add this method, since it isn't being used?

**Comment by lverns** [comment]
Except for the comment above, this looks good to merge.

File: server/src/main/java/umm3601/digitalDisplayGarden/PlantController.java (hunk 1 above)
Comment (frogrammer) [inline server/src/main/java/umm3601/digitalDisplayGarden/PlantController.java]: The method was used before we implemented `Comparator`. We can get rid of them.

**Comment by lverns** [comment]
Oh snap. I forgot about testing this. We swiped this code from a previous iteration, correct? If so, were there tests we can steal as well?

**Comment by frogrammer** [comment]
The code is from group [sn1999ec](https://github.com/UMM-CSci-3601-S17/digital-display-garden-iteration-4-sn1999ec). They do not have any tests, but I can add some tests tomorrow.

**Pull request merged**